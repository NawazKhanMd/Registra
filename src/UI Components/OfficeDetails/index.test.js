import React from 'react';
import renderer from 'react-test-renderer';
import { OfficeDetails } from './index';
import '@testing-library/jest-dom/extend-expect';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('react-router', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('firebase', () => ({
  initializeApp: () => ({
    analytics: jest.fn(),
    auth: jest.fn(),
  }), firestore: jest.fn(),
}));
jest.mock('react-redux', () => {
  const ActualReactRedux = require.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return 1;
    }),
    useDispatch: () => jest.fn()
  };
});

test('Stepper Loads', () => {
  const component = renderer.create(
    <OfficeDetails />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Simulating Form submit', () => {
  const wrapper = shallow((<OfficeDetails />));
  const fakeEvent = { preventDefault: jest.fn() };
  expect(wrapper.find('form')).toHaveLength(1);
  const entries = jest.fn()
  const get = (key) => {
    switch (key) {
      case "building_name":
        return "Mukti";
      case "city_area":
        return "Malleshwaram";
      case "landline_number":
        return "0802345225";
      case "building_add_1":
        return "169, Mukti";
      case "building_add_2":
        return "Malleshwaram, Bnglr";
      case "building_add_3":
        return "India, 560003";
      default:
        return "";
    }
  }
  const append = jest.fn()
  global.FormData = () => ({ entries, append, get })
  wrapper.find("form").simulate('submit', fakeEvent);
  expect(wrapper.find('form')).toHaveLength(1);
});

test('Simulating Back', () => {
  const wrapper = shallow((<OfficeDetails />));
  const fakeEvent = { preventDefault: jest.fn() };
  expect(wrapper.find('form')).toHaveLength(1);
  const entries = jest.fn()
  const get = (key) => {
    switch (key) {
      case "building_name":
        return "Mukti";
      case "city_area":
        return "Malleshwaram";
      case "landline_number":
        return "0802345225";
      case "building_add_1":
        return "169, Mukti";
      case "building_add_2":
        return "Malleshwaram, Bnglr";
      case "building_add_3":
        return "India, 560003";
      default:
        return "";
    }
  }
  const append = jest.fn()
  global.FormData = () => ({ entries, append, get })
  wrapper.find('.grey').simulate('click');
  expect(wrapper.find('form')).toHaveLength(1);
});

