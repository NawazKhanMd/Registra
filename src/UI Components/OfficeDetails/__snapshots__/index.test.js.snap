// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Loads 1`] = `
<div
  className="registration_container"
>
  <form
    autoComplete="off"
    name="OfficeForm"
    onSubmit={[Function]}
  >
    <div
      className="segment"
    >
      <h3>
        Office Info
      </h3>
    </div>
    <div
      className="d-flex f_row justify-content-evenly f-wrap"
    >
      <label>
        <input
          name="building_name"
          placeholder="Building Name"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          maxLength="10"
          name="landline_number"
          placeholder="Landline Number"
          required={true}
          size="10"
          type="text"
        />
      </label>
      <label>
        <input
          name="city_area"
          placeholder="City/Area"
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="segment"
    >
      <h3>
        Office Address
      </h3>
    </div>
    <div
      className="d-flex f_row justify-content-evenly f-wrap"
    >
      <label>
        <input
          name="building_add_1"
          placeholder="Street, Colony"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="building_add_2"
          placeholder="City, Country"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="building_add_3"
          placeholder="PO Box Number"
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="d-flex f_row justify-content-center"
    >
      <button
        className="grey  margin-10"
        onClick={[Function]}
      >
        Back
      </button>
      <button
        className="red margin-10"
        type="submit"
      >
        Next
      </button>
    </div>
  </form>
</div>
`;
